components/gitpod-protocol/go/reconnecting-ws.go

```diff
@@ -214,25 +224,20 @@ func (rc *ReconnectingWebsocket) connect(ctx context.Context) *WebsocketConnecti
 				return nil
 			}
 		}
-		var statusCode int
-		if resp != nil {
-			statusCode = resp.StatusCode
-		}
-		rc.log.WithField("url", rc.url).Info("websocket handshake")
 
+		delay := time.Duration(base + rand.Float64()*math.Min(cap, base*math.Pow(rc.reconnectionDelayGrowFactor, float64(attempt))))
 		rc.log.WithError(err).
 			WithField("url", rc.url).
 			WithField("badHandshakeCount", fmt.Sprintf("%d/%d", rc.badHandshakeCount, rc.badHandshakeMax)).
 			WithField("statusCode", statusCode).
-			Errorf("failed to connect, trying again in %d seconds...", uint32(delay.Seconds()))
+			WithField("attempt", attempt).
+			WithField("delay", delay.String()).
+			Error("failed to connect, trying again...")
 		select {
 		case <-rc.closedCh:
 			return nil
 		case <-time.After(delay):
-			delay = time.Duration(float64(delay) * rc.reconnectionDelayGrowFactor)
-			if delay > rc.maxReconnectionDelay {
-				delay = rc.maxReconnectionDelay
-			}
+			attempt++
```

Comment:
What if we used a package for this? It's way cleaner and offers the same configuration we need.
https://github.com/cenkalti/backoff/blob/v4.1.3/exponential.go#L54

Reply:
Can we do it in another PR? It takes time to test and I would like to focus on other things. I'm pretty sure this code works.

Reply:
Sure, up to you. Just feels the current version has higher risk of going wrong.

Reply:
I'm not sure what you mean. I've implemented Full Jitter from https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/ which seems to easies and perform better. 

The algorithm from https://github.com/cenkalti/backoff/blob/v4.1.3/exponential.go#L54 seems to be equal jitter based on randomness around moving interval. From the article above it performs worse since it is less random.

I can do a change tomorrow morning.

Reply:
Sorry, it was less about performance and more about using a tested package over our implementation which has higher chance of going wrong - we've not run it in anger in production.